spec/lib/tasks/populate_postcodes_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rake"
+require "spec_helper"
+
+Rails.application.load_tasks
+
+RSpec.describe Tasks do
+  describe "when running the task" do
+    before do
+      Postcode.destroy_all
+    end
+
+    let(:task) { Rake::Task["populate_postcodes"] }
+
+    it "should add postcodes in the database" do
+      expect(Postcode.count).to eq(0)
+
+      task.reenable
+      task.invoke("spec/lib/tasks/postcodes.csv")
+
+      expect(Postcode.count).to eq(3)
+    end
+
+    it "shouldn't fail when a postcode already exists" do
+      Postcode.create(postcode: "SE1 2TL")
+      expect(Postcode.count).to eq(1)
+
+      task.reenable
+      task.invoke("spec/lib/tasks/postcodes.csv")
+
+      expect(Postcode.count).to eq(3)
+    end
```

Comment:
I'd add one more test: that it doesn't overwrite any existing postcode rows.

e.g. `Postcode.create(postcode: "SE1 2TL", results: '{ "foo": "bar" }')` and assert that `the_postcode.results` is still the same after running the task.